File: reverse string.py
#reverse the string using two way pointer

s=list(input())
def reverseString( s: list[str]):  
        left, right = 0, len(s) - 1
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
reverseString(s)            
print("reverse string :",''.join(s) )

File: reverse_string.cpp
#include<iostream>
using namespace std;

int main(){
    string s;
    cin>>s;
    int left=0,right=s.length()-1;
    while (left < right){
       char temp=s[left];
       s[left]=s[right];
       s[right]=temp;
       left++;
       right--;    
    }
    cout<<s<<endl;  
}